internal/operators/authorino/authorino_operator.go

```diff
@@ -0,0 +1,134 @@
+	SubscriptionName: "authorino-operator",
+	TimeoutSeconds:   30 * 60,
+}
+
+// operator is an Authorino AI OLM operator plugin.
+type operator struct {
+	log       logrus.FieldLogger
+	config    *Config
+	templates *template.Template
+}
+
+// NewAuthorinoIOperator creates new Authorino operator.
+func NewAuthorinoOperator(log logrus.FieldLogger) *operator {
+	config := &Config{}
+	err := envconfig.Process(common.EnvConfigPrefix, config)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	templates, err := templating.LoadTemplates(templatesRoot)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	return &operator{
+		log:       log,
+		config:    config,
+		templates: templates,
+	}
+}
+
+// GetName reports the name of an operator.
+func (o *operator) GetName() string {
+	return Operator.Name
+}
+
+func (o *operator) GetFullName() string {
+	return "OpenShift AI"
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies(c *common.Cluster) (result []string, err error) {
+	return
+}
+
+// GetClusterValidationID returns cluster validation ID for the operator.
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDAuthorinoRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the operator.
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDAuthorinoRequirementsSatisfied)
+}
+
+// ValidateCluster checks if the cluster satisfies the requirements to install the operator.
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (result api.ValidationResult,
+	err error) {
+	result.ValidationId = o.GetClusterValidationID()
+	result.Status = api.Success
+	return
+}
+
+// ValidateHost returns validationResult based on node type requirements such as memory and CPU.
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host,
+	_ *models.ClusterHostRequirementsDetails) (result api.ValidationResult, err error) {
+	result.ValidationId = o.GetHostValidationID()
+	result.Status = api.Success
+	return
+}
+
+// GetProperties provides description of operator properties.
+func (o *operator) GetProperties() models.OperatorProperties {
+	return models.OperatorProperties{}
+}
+
+// GetMonitoredOperator returns the information that describes how to monitor the operator.
+func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
+	return &Operator
+}
+
+// GetHostRequirements provides the requirements that the host needs to satisfy in order to be able to install the
+// operator.
+func (o *operator) GetHostRequirements(ctx context.Context, cluster *common.Cluster,
+	host *models.Host) (result *models.ClusterHostRequirementsDetails, err error) {
+	preflightRequirements, err := o.GetPreflightRequirements(ctx, cluster)
+	if err != nil {
+		return
+	}
+	result = preflightRequirements.Requirements.Worker.Quantitative
```

Comment:
there are no resource required? do we have new pods in this operator ?

Reply:
There will be pods running, yes, at least the operator. I didn't find any documentation about the requirements. I will ask around.

Reply:
I asked about this, didn't get a response yet.

Reply:
if you get an update - please send a PR to update 
I will approve this to allow the project to proceed